Src/Support/Google.Apis.Auth/OAuth2/DirectUseExternalAccountCredential.cs

```diff
@@ -0,0 +1,181 @@
+﻿/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+using Google.Apis.Auth.OAuth2.Requests;
+using Google.Apis.Http;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Google.Apis.Auth.OAuth2
+{
+    /// <summary>
+    /// Base class for external account credentials used directly, i.e. without impersonation.
```

Comment:
Workload identity federation always involved _impersonating_ a service account. So when would an external credential ever be used _directly, without impersonation_?

Reply:
I wonder whether this is a matter of definition of impersonation. If someone uses DirectUseExternalAccountCredential, what principal is checked in the API? (I'd expect it to be *a* Google principal of some description.)

Reply:
Mmm. I think we could do with clarifying the AIP to be honest. I suspect "should be directly used without impersonation" should really be something like "should be used directly without a further explicit impersonation request to obtain a new token". I'm happy for the code to reflect the current AIP, and we can modify it if the AIP changes.

Reply:
OK, this thread is now going to become obsolete after upcoming changes based on the meeting with the Auth team. We'll only have ExternalAccountCredential (and the child classes for the specific credential types).